src/foam/mlang/expr/DateGroupingExpr.js

```diff
@@ -12,91 +12,69 @@ foam.CLASS({
   implements: [ 'foam.core.Serializable' ],
 
   documentation: `
-    An expr whose value is the name of a date grouping.
-    Currently works with hours, days, weeks, months, and years.
+    An expr whose value is the date/month/year of a date grouping.
   `,
 
   javaImports: [
     'foam.nanos.auth.CreatedAware',
     'java.util.Arrays',
     'java.util.Comparator',
-    'java.util.Date'
+    'java.util.Date',
+    'java.util.Calendar',
+    'java.util.Locale'
   ],
 
   properties: [
     {
       class: 'Enum',
       of: 'foam.mlang.expr.DateTypes',
       name: 'dateGroupingType',
-      value: 'DAYS'
-    },
-    {
-      class: 'FObjectArray',
-      of: 'foam.mlang.expr.DateGrouping',
-      name: 'dateGroups',
-      factory: function() {
-        return [];
-      },
-      javaFactory: `
-        return new foam.mlang.expr.DateGrouping[0];
-      `
+      value: 'MONTHS'
     }
   ],
 
   methods: [
     {
       name: 'f',
+      args: [
+        {
+          name: 'obj',
+          type: 'Object'
+        }
+      ],
       code: function(obj) {
-        var dateGroupsSorted = [
-          ...this.dateGroups
-        ];
-        
-        // TODO: Add a check for overlap between dateGroups and other data validation
-        dateGroupsSorted.sort((a, b) => a.low - b.high);
-
-        var objDiffFromTodayMs =  Math.floor(new Date().getTime()) - Math.floor(obj.created.getTime());
-
-        var objDiffFromTodayConverted = objDiffFromTodayMs / this.dateGroupingType.conversionFactorMs;
-        
-        var groupName = "Unknown Range";
-
-        dateGroupsSorted.forEach(group => {
-          if ( 
-            objDiffFromTodayConverted >= group.low && 
-            objDiffFromTodayConverted < group.high 
-          ) groupName = group.name;
-        })
-
-        return groupName;
+        if ( obj.created ) {
+          var dateCreated = new Date(obj.created);
+          if ( this.dateGroupingType == 'DAYS' ) {
```

Comment:
This should be comparing to the enum and not the string value like `foam.time.TimeUnit.DAY`